OSDOCS-3209 - Adding OSD wizard updates (#41560)
openshift/openshift-docs: modules/osd-create-cluster-red-hat-account.adoc

ifeval::["{context}" == "osd-getting-started"]
:osd-quickstart:
endif::[]
ifeval::["{context}" == "osd-creating-a-cluster-on-aws"]
:osd-on-aws:
endif::[]
ifeval::["{context}" == "osd-creating-a-cluster-on-gcp"]
:osd-on-gcp:
endif::[]

:_content-type: PROCEDURE
ifdef::osd-quickstart[]
[id="create-cluster-red-hat-account_{context}"]
= Creating a cluster with a Red Hat cloud account
endif::osd-quickstart[]
ifdef::osd-on-aws[]
[id="osd-create-aws-cluster-red-hat-account_{context}"]
= Creating a cluster on AWS with a Red Hat cloud account
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
[id="osd-create-aws-cluster-ccs_{context}"]
= Creating a cluster on GCP with a Red Hat cloud account
endif::osd-on-gcp[]

Through {OCM}, you can create an {product-title} cluster 
ifdef::osd-on-aws[]
on {AWS} 
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
on {GCP} 
endif::osd-on-gcp[]
using a standard cloud provider account owned by Red Hat.

.Procedure

. Log in to {console-redhat-com} and click *Create Cluster*.

. In the *Cloud* tab, click *Create Cluster* in the *Red Hat OpenShift Dedicated* row.

. In the *Billing model* dialog, configure the subscription type and infrastructure type:

> .. Select a subscription type.

Reviewer: The next step is "Select the *Red Hat cloud account* infrastructure type to deploy".  *Red Hat cloud account* infrastructure type only support subscription type "Annual". So it's better to just ask user select Annual subscription type.
And subscription type radio button will be hide once user have no quota, I wonder if we should clarify it here.

Author: @xueli181114 thank you for this comment. In the stage environment, I see two available subscription options for the Red Hat cloud account infrastructure type:

- Free trial (upgradeable)
- Annual: Fixed capacity subscription from Red Hat

I guess the available options depend on a customer's subscriptions. If I state only to select the "Annual" option, this might not cater for all use cases, such as the trial option. I have subsequently added the following note under the "Select a subscription type step":

> [NOTE]
> ====
> The subscription types that are available to you depend on your {product-title} subscriptions and resource quotas.
> ====

Reviewer: There is no plan to update this right?

Author: @xueli181114 I added the note instead. Based on the details outlined in my comment, if I change the step to only reference annual billing, then it might not cater for every possible use case. As I mentioned, I see **Free trial** in the UI at this point too. Is there something that I am missing?

Author: @xueli181114 following on from our discussion on Slack, I understand now that the trial option becomes unavailable when you opt to use a Red Hat cloud account. I have subsequently updated the step in this procedure as follows:

```
. Under *Billing model*, configure the subscription type and infrastructure type:
.. Select the *Annual* subscription type. Only the *Annual* subscription type is available when you deploy a cluster using a Red Hat cloud account.
+
[NOTE]
====
You must have the required resource quota for the *Annual* subscription type to be available. For more information, contact your sales representative or Red Hat support.
====
```

> .. Select a subscription type.

Reviewer: `Red Hat cloud account` only support subscription type `Annual` so it's better to clarify in the `Select a subscription type` step

Author: In the stage environment, I see two available subscription options for the Red Hat cloud account infrastructure type:

- Free trial (upgradeable)
- Annual: Fixed capacity subscription from Red Hat

I guess the available options depend on a customer's subscriptions. If I state only to select the "Annual" option, this might not cater for all use cases, such as the trial option. I have subsequently added the following note under the "Select a subscription type step":

> Note: The subscription types that are available to you depend on your OpenShift Dedicated subscriptions and resource quotas.

> .. Select the *Red Hat cloud account* infrastructure type to deploy {product-title} in a cloud provider account that is owned by Red Hat.

Reviewer 2: is there a reason we are not calling out CCS too?

.. Click *Next*.

ifdef::osd-quickstart[]
. Select *Run on Amazon Web Services* or *Run on Google Cloud Platform* to choose a cloud infrastructure provider 
endif::osd-quickstart[]
ifdef::osd-on-aws[]
. Select *Run on Amazon Web Services* 
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
. Select *Run on Google Cloud Platform* 
endif::osd-on-gcp[]
and click *Next*.

. On the *Cluster details* page, provide a name for your cluster and specify the cluster details:
.. Add a *Cluster name*.
.. Select a cluster version from the *Version* drop-down menu.
.. Select a cloud provider region from the *Region* drop-down menu.
.. Select a *Single zone* or *Multi-zone* configuration.

> .. Select a *Persistent storage* capacity and state the number of *Load balancers* that you require for your cluster.

Reviewer 2: I think we need to clarify the Load Balancers option here.  It is not clear to a user what they would put here.  They WILL be guessing.  We need to either link to docs that talk about what these load balancers are for, or tell them in the tool tip popup.  They will almost definitely be thinking "How many LB does this service use, where is the arch diagram?".  

These LBs are really for accessing the cluster over non-http/tcp ports.  For example, if they have a service running in the cluster on port 12345, then they would need at least one.

Author: Thank you for highlighting this. I have subsequently split the persistent storage and load balancer items into separate steps. I have also added references to the [Storage](https://docs.openshift.com/dedicated/osd_policy/osd-service-definition.html#sdpolicy-storage_osd-service-definition) and [Load balancers](https://docs.openshift.com/dedicated/osd_policy/osd-service-definition.html#load-balancers_osd-service-definition) sections in the OSD service definition from the steps and included links to those sections in the additional resources sections in both of the assemblies that use this module.

If further details are required in the docs in addition to what is in those sections, I recommend that we develop that as a separate work item outside of this PR.

> .. Leave *Enable user workload monitoring* selected to monitor your own projects in isolation from Red Hat Site Reliability Engineer (SRE) platform metrics. The option is enabled by default.

Reviewer 3: same suggestion: `s/The option/This option`

Author: Resolved.

.. Optional: Select *Enable additional etcd encryption* if you require etcd key value encryption. With this option, the etcd key values are encrypted, but not the keys. The option is in addition to the control plane storage encryption that encrypts the etcd volumes in {product-title} clusters by default.
+
[NOTE]
====
By enabling etcd encryption for the key values in etcd, you will incur a performance overhead of approximately 20%. The overhead is a result of introducing this second layer of encryption, in addition to the default control plane storage encryption that encrypts the etcd volumes. Consider enabling etcd encryption only if you specifically require it for your use case.
====
.. Click *Next*.

> . On the *Default machine pool* page, select a *Compute node instance type* and a *Compute node count*. The number and types of nodes that are available depend on your {product-title} subscription. If you are using multiple availability zones, the compute node count is per zone.

Reviewer: It's not "Compute node instance type". UI shows "Worker node instance type".
And it's also should be "Worker node count" rather than "Compute node count"

Author: I have suggested that the UI is updated to use "compute node" instead of "worker node" in https://issues.redhat.com/browse/HAC-451. I will keep this as is in the docs in preparation for the UI update.

+
[NOTE]
====

> After your cluster is created, you can change the number of compute nodes in your cluster, but you cannot change the compute node instance type. The number and types of nodes available to you depend on your {product-title} subscription.

Reviewer 2: This is saying that a cluster can only have 1 node type.  I am confirming this, but I know for CCS and ROSA you can add a new machine pool with different node types. Meaning that the node types have to be the same in the pool, but can have different types in different pools in the cluster

====

> . Optional: Expand *Edit node labels* to add labels to your nodes. Click *Add label* to add more node labels.

Reviewer: Need to add "Click Next button" to ask user to go to next step rather than just talk about privacy directly

Author: Thank you for highlighting this. Resolved.

> . Optional: Expand *Edit node labels* to add labels to your nodes. Click *Add label* to add more node labels.

Reviewer: Lack of step `Click Next` button

Author: Resolved.

. In the *Cluster privacy* dialog, select *Public* or *Private* to use either public or private API endpoints and application routes for your cluster.

> 

Reviewer: Need "Click Next button" step for the next page

Author: Thank you for highlighting this. Resolved.

. In the *CIDR ranges* dialog, configure custom classless inter-domain routing (CIDR) ranges or use the defaults that are provided.
+
[IMPORTANT]
====
CIDR configurations cannot be changed later. Confirm your selections with your network administrator before proceeding.

> If the cluster privacy is set to *Private*, you will not be able to access your cluster until you configure private connections in your cloud provider.

Reviewer 3: ```suggestion
If the cluster privacy is set to *Private*, you are not able to access your cluster until you configure private connections in your cloud provider.
```

Author: I have opted for "you cannot access your cluster" as a present tense replacement here. Thanks for highlighting this.

====

. On the *Cluster update strategy* page, configure your update preferences:
.. Choose a cluster update method. By selecting *Individual updates*, you can schedule each update individually. If you select *Recurring updates*, your cluster is updated on your preferred day and start time, when updates are available. *Individual updates* is the default option.
+
[NOTE]
====

> You can review the end-of-life dates in the update life cycle documentation for {product-title}. For more information, see _{product-title} update life cycle_.

Reviewer 3: Same comment about life cycle vs. lifecycle

Author: I will leave this as it is for now, for the reasons I outlined in the preceding comment thread.

====
+
.. If you opt for recurring updates, select a preferred day of the week and upgrade start time in UTC from the drop-down menus.
.. Optional: You can set a grace period for *Node Draining* during cluster upgrades. A *1 hour* grace period is set by default.
.. Click *Next*.
+
[WARNING]
====

> High and Critical security concerns (CVEs) will be patched automatically within 48 hours, regardless of your chosen update strategy.

Reviewer 3: ```suggestion
High and Critical security concerns (CVEs) are patched automatically within 48 hours, regardless of your chosen update strategy.
```

Author: Resolved.
